v2/init.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var errors = require('./errors');
+var read = require('../lib/read');
+var write = require('../lib/write');
+
+module.exports.Request = InitRequest;
+module.exports.Response = InitResponse;
+
+function InitRequest(version, hostPort, processName) {
+    if (!(this instanceof InitRequest)) {
+        return new InitRequest(version, hostPort, processName);
+    }
+    var self = this;
+    self.type = InitRequest.TypeCode;
+    self.version = version;
+    self.hostPort = hostPort;
+    self.processName = processName;
+}
+
+InitRequest.TypeCode = 0x01;
```

Comment:
No the codes are very much part of the protocol document and not chosen by this repository.

We could of course do the dup check in index... not sure there's a big win there.